core/src/com/unciv/models/Spy.kt

```diff
@@ -0,0 +1,164 @@
+package com.unciv.models
+
+import com.unciv.Constants
+import com.unciv.logic.IsPartOfGameInfoSerialization
+import com.unciv.logic.city.City
+import com.unciv.logic.civilization.Civilization
+import com.unciv.logic.civilization.NotificationCategory
+import com.unciv.logic.civilization.NotificationIcon
+import com.unciv.logic.civilization.managers.EspionageManager
+import kotlin.random.Random
+
+
+enum class SpyAction(val displayString: String) {
+    None("None"),
+    Moving("Moving"),
+    EstablishNetwork("Establishing Network"),
+    Surveillance("Observing City"),
+    StealingTech("Stealing Tech"),
+    RiggingElections("Rigging Elections"),
+    CounterIntelligence("Conducting Counter-intelligence")
+}
+
+
+class Spy() : IsPartOfGameInfoSerialization {
+    // `location == null` means that the spy is in its hideout
+    var location: String? = null
+    lateinit var name: String
+    var action = SpyAction.None
+        private set
+    var timeTillActionFinish = 0
```

Comment:
Instead of 'time' llease use 'turns' for consistency
'turnsRemainingToAction' or somesuch